tests/specialization_constants/specialization_constants_same_name_stress.h

```diff
@@ -0,0 +1,275 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Common checks for specialization constants same name stress test
+//
+*******************************************************************************/
+
+#ifndef __SYCLCTS_TESTS_SPEC_CONST_SAME_NAME_STRESS_H
+#define __SYCLCTS_TESTS_SPEC_CONST_SAME_NAME_STRESS_H
+
+#include "../common/common.h"
+#include "../common/once_per_unit.h"
+#include "specialization_constants_common.h"
+#include "specialization_constants_same_name_stress_helper.h"
+
+namespace specialization_constants_same_name_stress {
+using namespace sycl_cts;
+using namespace get_spec_const;
+
+template <typename T, typename via_kb>
+class kernel;
+
+template <typename T, typename via_kb>
+class check_specialization_constants_same_name_stress_for_type {
+  // Aliases for spec constants
+  static constexpr auto &sc_glob = ::sc_same_name<T>;
+  static constexpr auto &sc_out = g_outer::sc_same_name<T>;
+  static constexpr auto &sc_out_in = g_outer::g_inner::sc_same_name<T>;
+  static constexpr auto &sc_out_unn = g_outer::sc_ref_outer_unnamed<T>;
+  static constexpr auto &sc_out_unn_in =
+      g_outer::g_u_inner::sc_ref_outer_unnamed_inner<T>;
+  static constexpr auto &sc_out_unn_in_unn =
+      g_outer::g_u_inner::sc_ref_outer_unnamed_inner_unnamed<T>;
+  static constexpr auto &sc_unn = sc_ref_unnamed<T>;
+  static constexpr auto &sc_unn_out = u_outer::sc_ref_unnamed_outer<T>;
+  static constexpr auto &sc_unn_out_unn =
+      u_outer::sc_ref_unnamed_outer_unnamed<T>;
+  static constexpr auto &sc_unn_out_unn_in =
+      u_outer::u_inner::sc_ref_unnamed_out_unnamed_inner<T>;
+
+ public:
+  void operator()(util::logger &log, const std::string &type_name) {
+    using namespace spec_const_help;
+    try {
+      auto queue = util::get_cts_object::queue();
+      const sycl::context ctx = queue.get_context();
+      const sycl::device dev = queue.get_device();
+
+      if constexpr (via_kb::value) {
+        if (!dev.has(sycl::aspect::online_compiler)) {
+          once_per_unit::log(log, "Device does not support online compilation");
+          return;
+        }
+      }
+      // Size for result arrays
+      std::size_t size = to_integral(sc_st_id::SIZE);
+      sycl::range<1> range(size);
+      // Using malloc to not initialize for struct with no default constructor
+      // Array of expected default values
+      T *ref_def_values_arr = (T *)malloc(size * sizeof(T));
+      // Array of expected values
+      T *ref_arr = (T *)malloc(size * sizeof(T));
+      // Array for real default values
+      T *def_values_arr = (T *)malloc(size * sizeof(T));
+      // Array for real values
+      T *result_arr = (T *)malloc(size * sizeof(T));
```

Comment:
If this is a common usage, perhaps you could provide a class for this, like in https://github.com/keryell/teaching_C-plus-plus/tree/master/non-initializing

Reply:
Or just using a union is enough to avoid the initialization.

Reply:
Ronan, thanks for the advice to using `union`. I change using `malloc` to `unique pointer` in this commit: 929ccc5f226243b60c37af7d4e84e39116bf043f

Reply:
But actually if you use the `union` variable trick, there is no need to have the `unique_pointer` at all. Just use a local variable.

Reply:
using smart pointers was removed in this commit: 53bef60d6ec4b45f303bf4a538decbbc90d23a57